src/splitInteger.js

```diff
@@ -1,18 +1,24 @@
+'use strict';
 /**
  * @param {number} value
  * @param {number} numberOfParts
  *
  * @returns {number[]}
  */
+
 function splitInteger(value, numberOfParts) {
-  const parts = [];
+  let parts = [];
   let rest = value;
 
-  for (let partsLeft = numberOfParts; partsLeft > 0; partsLeft--) {
-    const part = Math.floor(rest / partsLeft);
+  if (value > numberOfParts) {
+    for (let partsLeft = numberOfParts; partsLeft > 0; partsLeft--) {
+      const part = Math.floor(rest / partsLeft);
 
-    parts.push(part);
-    rest -= part;
+      parts.push(part);
+      rest -= part;
+    }
+  } else {
+    parts = Array(numberOfParts).fill(0);
   }
```

Comment:
You shouldn't modify the solution: the arguments are always valid, so you don't need to add this.